docs/developers.md

```diff
@@ -0,0 +1,275 @@
+## Zenbot exchange API
+This document is written to help developers implement new extensions for Zenbot.
+
+It is reverse engineered from inspecting the Zenbot files and the GDAX extension and is not a definitive guide for developing an extension.
+
+Any contribution that makes this document better is certainly welcome.
+ 
+The document is an attempt to describe the interface functions used for communication with an exchange and a few helper functions. Each function has a set of calling parameters and return values and statuses
+
+The input parameters are packed in the "opts" object, and the results from invoking the function are returned in an object.
+
+## Error handling
+
+Errors are returned to calling program through a callback functon of this form:
+```javascript
+cb (err)
+```
+The expected content of "err" is as follows:
+```javascript
+  { code: 'HTTP_STATUS', body: body }
+```
+
+**Non recoverable errors** should be handled by the actual extension function. A typical error is "Page not found", which most likely is caused by a malformed URL. Such errors should return a descriptive message and force a program exit.
+
+**Recoverable errors** affecting trades should be handled by zenbot, while others could be handled in the extension layer. This needs to be clarified.
+
+
+Some named errors are already handled by the main program (see getTrades below). These are:
+```
+  'ETIMEDOUT', // possibly recoverable 
+  'ENOTFOUND', // not recoverable (404?)
+  'ECONNRESET' // possibly recoverable
+```
+Zenbot may have some GDAX-specific code. In particular that pertains to return values from exchange functions. Return values in general should be handled in a exchange agnostic and standardized way to make it easiest possible to write extensions.
+
+**Some variables in the "exchange" object are worth mentioning**
+```
+  name: 'some_exchange_name'
+  historyScan: 'forward', 'backward' or false
+  makerFee: fee_from_exchange (numeric)
```

Comment:
recently added `takerFee`, but https://github.com/carlos8f/zenbot/issues/237 might change that soon. Might be worth mentioning.